code/modules/mod/mod_core.dm

```diff
@@ -355,3 +355,77 @@
 	add_charge(uses_needed * charge_given)
 	balloon_alert(user, "core refueled")
 	return TRUE
+
+/obj/item/mod/core/plasma/lavaland
+	name = "MOD plasma flower core"
+	icon_state = "mod-core-plasma-flower"
+	desc = "A strange flower from the desolate wastes of lavaland. It pulses with a strange purple glow.  \
+		The wires coming out of it could be hooked into a MODsuit."
+	light_system = MOVABLE_LIGHT
+	light_color = "#cc00cc"
+	light_range = 2
+
+	// Slightly better than the normal plasma core.
+	// Not super sure if this should just be the same, but will see.
+	maxcharge = 15000
+	charge = 15000
+
+	/// The mob to be spawned by the core
+	var/mob/living/spawned_mob_type = /mob/living/basic/butterfly/lavaland/temporary
+	/// Max number of mobs it can spawn
+	var/max_spawns = 3
+
+	/// Mob spawner for the core
+	var/datum/component/spawner/mob_spawner
+
+
+/obj/item/mod/core/plasma/lavaland/Destroy()
+	if(mod?.wearer)
+		mod.wearer.particles = null
+	return ..()
+
+/obj/item/mod/core/plasma/lavaland/proc/new_mob(spawner, mob/living/basic/butterfly/lavaland/temporary/spawned)
+	SIGNAL_HANDLER
+	if(spawned)
+		spawned.source = src
+
+/obj/item/mod/core/plasma/lavaland/proc/on_toggle()
+	SIGNAL_HANDLER
+	if(mod.active)
+		START_PROCESSING(SSprocessing, src)
+		mod.wearer.particles = new /particles/pollen()
+		mob_spawner = mod.wearer.AddComponent(/datum/component/spawner, spawn_types=list(spawned_mob_type), spawn_time=5 SECONDS, max_spawned=3)
+		RegisterSignal(mob_spawner, COMSIG_SPAWNER_SPAWNED, PROC_REF(new_mob))
+		RegisterSignal(mod.wearer, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(flowas))
+
+	else
+		STOP_PROCESSING(SSprocessing, src)
+		mod.wearer.particles = null
+		UnregisterSignal(mob_spawner, COMSIG_SPAWNER_SPAWNED)
+		UnregisterSignal(mod.wearer, COMSIG_MOVABLE_PRE_MOVE)
+		for(var/datum/mob in mob_spawner.spawned_things)
+			qdel(mob)
+		qdel(mob_spawner)
+
+/obj/item/mod/core/plasma/lavaland/install(obj/item/mod/control/mod_unit)
+	. = ..()
+	RegisterSignal(mod_unit, COMSIG_MOD_TOGGLED, PROC_REF(on_toggle))
+
+/obj/item/mod/core/plasma/lavaland/uninstall(obj/item/mod/control/mod_unit)
+	. = ..()
+	UnregisterSignal(mod_unit, COMSIG_MOD_TOGGLED)
+
+/obj/item/mod/core/plasma/lavaland/proc/flowas(mob/living/wearer)
+	SIGNAL_HANDLER
+	var/static/list/possible_flower_types = list(
+		/obj/structure/flora/bush/lavendergrass/style_random,
+		/obj/structure/flora/bush/flowers_yw/style_random,
+		/obj/structure/flora/bush/flowers_br/style_random,
+		/obj/structure/flora/bush/flowers_pp/style_random
+	)
+	var/chosen_type
+	if(!chosen_type || prob(5))
+		chosen_type = pick(possible_flower_types)
```

Comment:
I don't understand this bit, ``chosen_type`` will never be already set here, why do we have it? And what's the prob(5) supposed to do?